app/workers/metrics/record_daily_usage_worker.rb

```diff
@@ -4,14 +4,44 @@ class RecordDailyUsageWorker
     sidekiq_options queue: :low_priority, retry: 10
 
     def perform
-      articles_min_15_score_past_24h = Article.published.where("score >= 15").size
+      articles_min_15_score_past_24h = Article.published.where("score >= ? AND published_at > ?", 15, 1.day.ago).size
       DataDogStatsClient.count("articles.min_15_score_past_24h", articles_min_15_score_past_24h, tags: { resource: "articles" })
 
-      first_articles_past_24h = Article.where(nth_published_by_author: 1).size
+      first_articles_past_24h = Article.where(nth_published_by_author: 1).where("published_at > ?", 1.day.ago).size
       DataDogStatsClient.count("articles.first_past_24h", first_articles_past_24h, tags: { resource: "articles" })
 
       new_users_min_1_comment_past_24h = User.where("comments_count >= ? AND created_at > ?", 1, 24.hours.ago).size
       DataDogStatsClient.count("users.new_min_1_comment_past_24h", new_users_min_1_comment_past_24h, tags: { resource: "users" })
+
+      nagative_reactions_past_24h = Reaction.where("points < 0").where("created_at > ?", 1.day.ago).size
+      DataDogStatsClient.count("reactions.negative_past_24h", nagative_reactions_past_24h, tags: { resource: "reactions" })
+
+      reports_past_24_hours = FeedbackMessage.where(category: ["spam", "other", "rude or vulgar", "harassment"]).where("created_at > ?", 1.day.ago).size
+      DataDogStatsClient.count("feedback_messages.reports_past_24_hours", reports_past_24_hours, tags: { resource: "feedback_messages" })
+
+      get_days_active_past_week_counts
+    end
+
+    private
+
+    def get_days_active_past_week_counts
+      ids_by_day = []
+      7.times do |i|
+        ids_by_day << PageView.where("created_at > ? AND created_at < ?", (i + 1).days.ago, i.days.ago).where.not(user_id: nil).pluck(:user_id).uniq
+      end
+      non_new_user_ids = User.where("created_at < ?", 7.days.ago).where(id: ids_by_day.flatten).pluck(:id)
```

Comment:
`ids_by_day.flatten.uniq` here and below would likely shrink this array a bit for the lookup. You could also remove the `non_new_user_ids` before you run the second lookup. 

I worry about this approach scaling but I'm ok with this for now. Being a metric job, if it gets slow or times out it won't really affect anything else but something to keep in mind.

Reply:
I've run this in the console and it's a reasonably quick algorithm at our current scale. Since it's only running once a day or so, I think it should work for us for a good while.

Good call on the `uniq`.